pkg/rotator/rotator.go

```diff
@@ -147,20 +147,22 @@ type SyncingReader interface {
 
 // CertRotator contains cert artifacts and a channel to close when the certs are ready.
 type CertRotator struct {
-	reader                 SyncingReader
-	writer                 client.Writer
-	SecretKey              types.NamespacedName
-	CertDir                string
-	CAName                 string
-	CAOrganization         string
-	DNSName                string
-	IsReady                chan struct{}
-	Webhooks               []WebhookInfo
-	RestartOnSecretRefresh bool
-	certsMounted           chan struct{}
-	certsNotMounted        chan struct{}
-	wasCAInjected          *atomic.Bool
-	caNotInjected          chan struct{}
+	reader                      SyncingReader
+	writer                      client.Writer
+	SecretKey                   types.NamespacedName
+	CertDir                     string
+	CAName                      string
+	CAOrganization              string
+	DNSName                     string
+	IsReady                     chan struct{}
+	Webhooks                    []WebhookInfo
+	RestartOnSecretRefresh      bool
+	EnableExtKeyUsageClientAuth bool
```

Comment:
Rather than wrapping this, do we want to allow consumers to just supply a list of ExtendedKeyUsages that overrides the default of just server auth? That would avoid needing to expand things further if more uses are encountered.

We'd need to probably use a pointer to differentiate between "unset" and "intentionally empty"